src/mds/nameserver2/curvefs.cpp

```diff
@@ -280,6 +282,49 @@ StatusCode CurveFS::CreateFile(const std::string & fileName,
         }
     }
 
+    std::vector<PoolIdType> logicalPools =
+                topology_->GetLogicalPoolInCluster();
+    logicalPools = topology_->GetLogicalPoolInCluster();
+    if (0 == logicalPools.size() && length != 0) {
+        LOG(ERROR) << "[ChooseSingleLogicalPool]:"
+                << " Does not have any available logicalPools.";
+        return StatusCode::kFileLengthNotSupported;
+    }
+
+    bool EnoughFlag = false;
+    for (auto pid : logicalPools) {
+    LogicalPool lPool;
+    if (!topology_->GetLogicalPool(pid, &lPool)) {
```

Comment:
The following code from lines 296 to 312 has the same logic as TopologyChunkAllocatorImpl::ChooseSingleLogicalPool. You can consider putting this part of the logic in topology_, such as encapsulating a GetRemainingSpaceInLogicalPool()